lib/src/home/contacts/add_contacts_page.dart

```diff
@@ -0,0 +1,237 @@
+        keyboardType: TextInputType.text,
+        inputFormatters: [LengthLimitingTextInputFormatter(45)],
+        controller: _contactFirstName,
+        autofocus: true,
+        decoration: const InputDecoration(
+            labelText: Strings.firstNameLabel, icon: Icon(Icons.person)),
+        validator: Validations.nameValidation);
+
+    TextFormField inputLastName = TextFormField(
+        key: Keys.inputLastNameKey,
+        textAlignVertical: TextAlignVertical.center,
+        keyboardType: TextInputType.text,
+        inputFormatters: [LengthLimitingTextInputFormatter(45)],
+        controller: _contactLastName,
+        decoration: const InputDecoration(
+            labelText: Strings.lastNameLabel, icon: Icon(Icons.person)),
+        validator: Validations.nameValidation);
+
+    TextFormField inputPhoneNumber = TextFormField(
+      key: Keys.inputPhoneNumberKey,
+      textAlignVertical: TextAlignVertical.center,
+      keyboardType: TextInputType.phone,
+      inputFormatters: [FilteringTextInputFormatter.allow(RegExp(r'[0-9\+]*'))],
+      controller: _contactPhoneNumber,
+      maxLength: 16,
+      validator: (_) {
+        return Validations.phoneNumberValidation(_contactPhoneNumber.unmasked);
+      },
+      decoration: const InputDecoration(
+        labelText: Strings.phoneNumberLabel,
+        icon: Icon(Icons.phone),
+      ),
+    );
+
+    final picture = SizedBox(
+        width: 120.0,
+        height: 120.0,
+        child: GestureDetector(
+          onTap: () async {
+            showDialog(
+                context: context,
+                builder: (BuildContext context) =>
+                    ImageChooserDialog(onImagePick: (image) {
+                      setState(() {
+                        _contactPhoto = image;
+                      });
+                    }));
+          },
+          child: CircleAvatar(
+              child: ImageWithPlaceholderIcon(
+                  key: Keys.avatarKey, image: _contactPhoto)),
+        ));
+
+    ListView content = ListView(
+      padding: const EdgeInsets.symmetric(vertical: 20.0, horizontal: 10.0),
+      children: <Widget>[
+        const SizedBox(height: 20),
+        Form(
+          key: _formKey,
+          child: Column(
+            children: <Widget>[
+              picture,
+              inputFirstName,
+              inputLastName,
+              inputPhoneNumber
+            ],
+          ),
+        ),
+      ],
+    );
+
+    return Scaffold(
+      appBar: AppBar(
+        leading: IconButton(
+          icon: const Icon(Icons.close),
+          onPressed: () {
+            Navigator.of(context).pop();
+          },
+        ),
+        title: const Text(Strings.addContactPageTitle),
+        actions: <Widget>[
+          SizedBox(
+            width: 80,
+            child: IconButton(
+              icon: const Text(
+                'Save',
+                style: TextStyle(fontWeight: FontWeight.bold),
+              ),
```

Comment:
Don't give text for an icon button 
Either use elevated/outline button for text 
Or use a save icon inside icon button